src/server/data_store/lru_cache_data_store.cc

```diff
@@ -0,0 +1,51 @@
+#include "lru_cache_data_store.h"
+
+::std::string KeyValuePair::get_key() const {
+  return key;
+}
+
+::std::string KeyValuePair::get_value() const {
+  return value;
+}
+
+// This is not thread safe and very naive implementation.
+::LruCacheDataStore::LruCacheDataStore(::std::uint32_t capacity) {
+  capacity = capacity;
+  key_iterator_map = ::std::unordered_map<::std::string, ::std::list<KeyValuePair>::iterator>();
+  pairs = ::std::list<KeyValuePair>();
+}
+
+::LruCacheDataStore::~LruCacheDataStore() {}
+
+::std::string LruCacheDataStore::Get(::std::string key) {
+  if (key_iterator_map.find(key) == key_iterator_map.end()) {
+    return nullptr;
```

Comment:
yeah, I see your point about returning Optional<value>

Reply:
There's https://en.cppreference.com/w/cpp/utility/optional . I will probably change it soon, want to study on best practice or common pattern for this type of thing in cpp first.